core/src/main/scala/filodb.core/binaryrecord2/RecordSchema.scala

```diff
@@ -144,33 +152,35 @@ final class RecordSchema(val columnTypes: Seq[Column.ColumnType],
     asZCUTF8Str(UnsafeUtils.ZeroPointer, address, index)
 
   /**
-   * EXPENSIVE. Creates a easy-to-read Java String representation of the contents of this BinaryRecord.
+   * EXPENSIVE to do at server side. Creates a easy-to-read string
+   * representation of the contents of this BinaryRecord.
    */
-  def stringify(base: Any, offset: Long): String = {
-    import Column.ColumnType._
-    val parts: Seq[Any] = columnTypes.zipWithIndex.map {
-      case (IntColumn, i)    => getInt(base, offset, i)
-      case (LongColumn, i)   => getLong(base, offset, i)
-      case (DoubleColumn, i) => getDouble(base, offset, i)
-      case (StringColumn, i) => asJavaString(base, offset, i)
-      case (TimestampColumn, i) => getLong(base, offset, i)
-      case (BitmapColumn, i) => getInt(base, offset, i) != 0
-      case (MapColumn, i)    =>
-        val consumer = new StringifyMapItemConsumer
-        consumeMapItems(base, offset, i, consumer)
-        consumer.prettyPrint
-      case (PartitionKeyColumn, i)    =>
-        // FIXME: Looks like we are assuming that partition key is always a map. This is not necessarily true
-        val consumer = new StringifyMapItemConsumer
-        consumeMapItems(base, offset, i, consumer)
-        consumer.prettyPrint
-    }
-    s"b2[${parts.mkString(",")}]"
-  }
+  def stringify(base: Any, offset: Long): String =
+    s"b2[${mapify(base, offset).map(_.productIterator.mkString("=")).mkString(",")}]"
 
   def stringify(address: NativePointer): String = stringify(UnsafeUtils.ZeroPointer, address)
   def stringify(bytes: Array[Byte]): String = stringify(bytes, UnsafeUtils.arayOffset)
 
+  /**
+    * EXPENSIVE to do at server side. Creates a stringified map with contents of this BinaryRecord.
+    */
+  def mapify(base: Any, offset: Long): Map[String, String] = {
```

Comment:
IMO mapify() is not the same output as stringify().  Once you do this, then non-map items will get flattened along with the map key-values, and it makes it harder to debug the layout.  Can we please keep both versions?  Or at the very least, delineate where the maps start and end so that it's easy to see which elements are part of the map, and which are not.

Reply:
Another problem is that maps do not preserve order.  The entire point of prettyPrint is that it must print out fields in order and we must be able to tell what is in each field, for easy debugging.
I think what you can do is keep most of stringify().  For mapColumns, use mapify().  For PartitionKeyColumns, call stringify recursively on itself.  Then the layout is clear.

Reply:
Sorry, not pretty print, stringify.

Reply:
The intention of this flattening together is for binary record processing details to be abstracted within. We don't want clients to be processing columns separately depending on the schema. The usage of `mapify` would for example be to resolve the partKey into a bunch of key-value pairs such as to respond to a prometheus api. We dont want caller to act situationally based on the schema. One piece of code should apply to all cases. If there is a use case for unflattened map, then we should have another API.

For preserving order, we can certainly do Seq[Tuple2] instead of map. That may be more appropriate.